src/modules/airspeed_selector/airspeed_selector_main.cpp

```diff
@@ -302,9 +302,9 @@ AirspeedModule::Run()
 			_airspeed_validator[i].update_airspeed_validator(input_data);
 
 		}
-	}
 
-	select_airspeed_and_publish();
+		select_airspeed_and_publish();
```

Comment:
select_airspeed_and_publish() should also be executed if no airspeed sensor is connected, so with this change I think it wouldn't work.
Inside select_airspeed_and_publish, _airspeed_validator is used yes, but only if the number of airspeed sensors is >0, in which case _airspeed_validator is != nullptr. Probably not the nicest way to program it, but should be safe, no?

Reply:
Aha. However, there is still one catch: if `new` fails to allocate because you're out of RAM `_airspeed_validator` is also null even though `_number_of_airspeed_sensors` is `> 0`. We need some kind of mechanism to check for that failure case and notify that airspeed is broken and we must not fly.